localstack/services/lambda_/resource_providers/aws_lambda_url.py

```diff
@@ -0,0 +1,132 @@
+# LocalStack Resource Provider Scaffolding v2
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+
+
+class LambdaUrlProperties(TypedDict):
+    AuthType: Optional[str]
+    TargetFunctionArn: Optional[str]
+    Cors: Optional[Cors]
+    FunctionArn: Optional[str]
+    FunctionUrl: Optional[str]
+    InvokeMode: Optional[str]
+    Qualifier: Optional[str]
+
+
+class Cors(TypedDict):
+    AllowCredentials: Optional[bool]
+    AllowHeaders: Optional[list[str]]
+    AllowMethods: Optional[list[str]]
+    AllowOrigins: Optional[list[str]]
+    ExposeHeaders: Optional[list[str]]
+    MaxAge: Optional[int]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class LambdaUrlProvider(ResourceProvider[LambdaUrlProperties]):
+    TYPE = "AWS::Lambda::Url"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[LambdaUrlProperties],
+    ) -> ProgressEvent[LambdaUrlProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/FunctionArn
+
+        Required properties:
+          - TargetFunctionArn
+          - AuthType
+
+        Create-only properties:
+          - /properties/TargetFunctionArn
+          - /properties/Qualifier
+
+        Read-only properties:
+          - /properties/FunctionUrl
+          - /properties/FunctionArn
+
+        IAM permissions required:
+          - lambda:CreateFunctionUrlConfig
+
+        """
+        model = request.desired_state
+        lambda_client = request.aws_client_factory.lambda_
+
+        response = lambda_client.create_function_url_config(
+            Qualifier=model["Qualifier"],
+            Cors=model["Cors"],
```

Comment:
Qualifier and Cors are optional parameters so it's possible python will raise an exception here

Reply:
I'll make a changes to it.